### ISSUE TITLE
Account settings missing in Admin CP

### ISSUE THREAD

**Issue by PaulBender** [mybb/mybb#1821]
The account settings **Receive PM notifications for new buddy requests.** and **Automatically accept buddy requests (if the above checkbox is ticked, a PM is sent informing of the new buddy connection)** are not available when editing a user in the Admin CP.

**Comment by Sama34** [comment]
`sourceeditor` needs to be updated during upgrade now.

**Comment by Destroy666x** [comment]
> `sourceeditor` needs to be updated during upgrade now.

What exactly do you want to update on upgrade and why?

**Comment by Sama34** [comment]
> What exactly do you want to update

The `sourceeditor` value of already existing users.

> and why

Because the default value for `sourceeditor` was modified in this issue.

**Comment by Destroy666x** [comment]
Umm... That doesn't make any sense. If someone has changed the option to 0 by himself you want to force updating it to 1? How does changing the default value in next version require changing it for all current users..?

**Comment by Sama34** [comment]
A change was applied in this issue, such change was unnecessary to fix this issue and should had been discussed in the proper issue. #1393

**Comment by JN-Jones** [comment]
I'll look at the changes/issues/whatever later. It's 2am here and I'm too tired now...

**Comment by Destroy666x** [comment]
@Sama34, in that issue you say:

> Default value should be off (0) as that is how we introduced it.

And that's incorrect, which caused problems - as you can see, it was `1` by default here (and `0` everywhere else) so it was changed to correct default `0` - I don't see what's wrong. It has nothing to do with the linked issue either because it changes the default behavior without any setting and a setting for it can still be added if needed..

**Comment by JN-Jones** [comment]
Ok, so:
There's nothing to update on the upgrade here @Sama34 This fixes only a wrong default value in one place - if an user hasn't changed their value till now they likely want it like that (or don't know that there's an option but changing it doesn't make sense here).
However issue #1393 will likely change the line once again (introducing an admin option as default) so yes, maybe it would have been better to make the change there. But as said the potential fix for that issue will change the line anyways and isn't affected by this change (it won't break anything for you). So there isn't the need to revert that line again just to change it there too. Also remember that it may happen that the issue won't be fixed in 1.8.5 and in that case the correct default value is used now.

tl;dr: Leave it as it is now and change it when you fix #1393

**Comment by Sama34** [comment]
The thing is it was already planed to run the upgrade script.
http://community.mybb.com/thread-158793-post-1100254.html#pid1100254

But well..

**Comment by JN-Jones** [comment]
As said above: this issue/line is a different one and there's absolutely no need to change anything in the upgrade script for this change. The upgrade script is required for the other issue but as said above too that should be changed in that issue. Fixing/changing half of the issue here and the other half later (maybe even in a later version) would cause more issues than it would fix.

Again: this issue and the other issue are independent, they only change the same line. This issue fixes only a wrong default value which may have lead to wrong behavior while the other issue will add a new setting to change that default behavior. First means we don't need to change existing values, second means we need to change them.
